### PR TITLE
Lesson 5-7 part 1-3:Chardev +procfs+ sysfs

### PR DESCRIPTION
**Pull request by MrStamp**
Implementation of simple module

### CODE REVIEW

**Comment by galkinletter** [comment]
chrdev.c - static int count = 1;  Never used.

Please comment in  driver source.

**Review by galkinletter — CHANGES_REQUESTED** [review]
chrdev.c - static int count = 1; Never used.
Please comment in driver source.

File: 02-KernelModules_CharacterDevice/chrdev.c [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+	}
+	pr_info("chrdev: device node created successfully\n");
+
+	example_kobject = kobject_create_and_add(DEVICE_NAME, kernel_kobj);
+    	if(!example_kobject) {
+        pr_debug("failed to create kobject \n");
+        return -ENOMEM;
+    	}
+    	error = sysfs_create_file(example_kobject, &sysfs_attribute.attr);
+    	if (error) {
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/chrdev.c]: Error handling is broken here.

File: 02-KernelModules_CharacterDevice/chrdev.c [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+#define DEVICE_NAME	"chrdev_example"
+
+static int error;
+static int BUFFER_SIZE=1024;
+static int major = 0; 
+static int minor = 0; 
+//static int count = 1; 
+module_param(major, int, S_IRUGO);
+module_param(minor, int, S_IRUGO);
+module_param(count, int, S_IRUGO);
```
Comment (oleg-h2o) [inline 02-KernelModules_CharacterDevice/chrdev.c]: Unused parameter.

**Review by oleg-h2o — CHANGES_REQUESTED** [review]
Plz fix. 
There are indentation problems in the file also.

File: 02-KernelModules_CharacterDevice/Makefile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+obj-m += chrdev.o
+
+BUILD_KERNEL=/lib/modules/$(shell uname -r)/build
+
+all:
+	make -C $(BUILD_KERNEL) M=$(PWD) modules
+
+clean:
+	make -C $(BUILD_KERNEL) M=$(PWD) clean
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/Makefile:9]: Please add empty line at the end of file
```suggestion
	make -C $(BUILD_KERNEL) M=$(PWD) clean
	
```

**Review by galkinletter — CHANGES_REQUESTED** [review]
Please fix:

- Error handling
- Code style
- Unused parameter

File: 02-KernelModules_CharacterDevice/chrdev.c [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+//procfs
+
+static int dev_proc_show(struct seq_file *m, void *v) {
+  seq_printf(m, "Buffer size = %d\nData length = %d\n", BUFFER_SIZE, data_size);
+  return 0;
+}
+
+static int dev_proc_open(struct inode *inode, struct  file *file) {
+  return single_open(file, dev_proc_show, NULL);
```
Comment (galkinletter) [inline 02-KernelModules_CharacterDevice/chrdev.c]: Please Check kernel code with checkpatch and cppcheck

**Review by galkinletter — CHANGES_REQUESTED** [review]
Please Check kernel code with checkpatch and cppcheck

**Comment by galkinletter** [comment]
It can be seen that the work was done

**Pull request merged**